com.unity.render-pipelines.universal/Editor/ShaderGraph/Includes/Varyings.hlsl

```diff
@@ -105,7 +105,9 @@ Varyings BuildVaryings(Attributes input)
 #endif
 
 #ifdef VARYINGS_NEED_VIEWDIRECTION_WS
-    output.viewDirectionWS = GetWorldSpaceNormalizeViewDir(positionWS);
+    // Need the unnormalized direction here as otherwise interpolation is incorrect.
+    // It is normalized after interpolation in the fragment shader.
+    output.viewDirectionWS = GetWorldSpaceViewDir(positionWS);
```

Comment:
btw, not something for this PR but as awareness in URP we now compute view direction always in fragment https://github.com/Unity-Technologies/Graphics/pull/3099

Reply:
Thanks!